web-backend/src/test/java/com/statusneo/vms/service/OtpServiceTest.java

```diff
@@ -0,0 +1,212 @@
+
+/*
+ * Copyright [2025] StatusNeo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Author: StatusNeo
+ */
+
+package com.statusneo.vms.service;
+
+import com.statusneo.vms.model.Email;
+import com.statusneo.vms.model.Otp;
+import com.statusneo.vms.repository.OtpRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class OtpServiceTest {
+
+    @InjectMocks
+    private OtpService otpService;
+
+    @Mock
+    private OtpRepository otpRepository;
+
+    @Mock
+    private EmailService emailService;
+
+    @BeforeEach
+    void setUp() throws Exception{
```

Comment:
Why is this needed? If passing in subject is necessary then just use ctor injection instead of field.